From a751aacef0ada75fc295cbdc6479482f45f8899b Mon Sep 17 00:00:00 2001
From: Shawn Ferry <sferry@gmail.com>
Date: Fri, 14 Apr 2017 13:18:47 -0400
Subject: [PATCH 4/5] (no-upstream) install path changes

Patches not suitable for upstream
25232171 puppet hard codes new path locations
25232111 puppet install.rb vardir argument has no impact
---
 install.rb                                    |  8 ++++----
 lib/puppet/defaults.rb                        |  2 +-
 lib/puppet/face/config.rb                     |  2 +-
 lib/puppet/face/module/install.rb             | 17 +++++++++--------
 lib/puppet/face/module/list.rb                | 10 +++++-----
 lib/puppet/face/module/uninstall.rb           |  4 ++--
 lib/puppet/provider/package/puppet_gem.rb     |  2 +-
 lib/puppet/reference/configuration.rb         |  2 +-
 lib/puppet/util/run_mode.rb                   |  2 +-
 man/man8/puppet-config.8                      |  2 +-
 spec/unit/provider/package/puppet_gem_spec.rb |  2 +-
 spec/unit/util/run_mode_spec.rb               |  4 ++--
 14 files changed, 38 insertions(+), 37 deletions(-)

diff --git a/install.rb b/install.rb
index cad3ad8dc..abc8cf234 100755
--- a/install.rb
+++ b/install.rb
@@ -193,7 +193,7 @@ def prepare_installation
     opts.on('--codedir[=OPTIONAL]', 'Installation directory for code files', 'Default /etc/puppetlabs/code') do |codedir|
       InstallOptions.codedir = codedir
     end
-    opts.on('--vardir[=OPTIONAL]', 'Installation directory for var files', 'Default /opt/puppetlabs/puppet/cache') do |vardir|
+    opts.on('--vardir[=OPTIONAL]', 'Installation directory for var files', 'Default /var/puppetlabs/puppet/cache') do |vardir|
       InstallOptions.vardir = vardir
     end
     opts.on('--rundir[=OPTIONAL]', 'Installation directory for state files', 'Default /var/run/puppetlabs') do |rundir|
@@ -205,7 +205,7 @@ def prepare_installation
     opts.on('--bindir[=OPTIONAL]', 'Installation directory for binaries', 'overrides RbConfig::CONFIG["bindir"]') do |bindir|
       InstallOptions.bindir = bindir
     end
-    opts.on('--localedir[=OPTIONAL]', 'Installation directory for locale information', 'Default /opt/puppetlabs/puppet/share/locale') do |localedir|
+    opts.on('--localedir[=OPTIONAL]', 'Installation directory for locale information', 'Default /usr/puppetlabs/puppet/share/locale') do |localedir|
       InstallOptions.localedir = localedir
     end
     opts.on('--ruby[=OPTIONAL]', 'Ruby interpreter to use with installation', 'overrides ruby used to call install.rb') do |ruby|
@@ -291,7 +291,7 @@ def prepare_installation
   elsif $operatingsystem == "windows"
     vardir = File.join(Dir::COMMON_APPDATA, "PuppetLabs", "puppet", "cache")
   else
-    vardir = "/opt/puppetlabs/puppet/cache"
+    vardir = "/var/puppetlabs/puppet/cache"
   end
 
   if not InstallOptions.rundir.nil?
@@ -322,7 +322,7 @@ def prepare_installation
     if $operatingsystem == "windows"
       localedir = File.join(Dir::PROGRAM_FILES, "Puppet Labs", "Puppet", "puppet", "share", "locale")
     else
-      localedir = "/opt/puppetlabs/puppet/share/locale"
+      localedir = "/usr/puppetlabs/puppet/share/locale"
     end
   end
 
diff --git a/lib/puppet/defaults.rb b/lib/puppet/defaults.rb
index 2af113634..609950841 100644
--- a/lib/puppet/defaults.rb
+++ b/lib/puppet/defaults.rb
@@ -1256,7 +1256,7 @@ EOT
       :desc       => "File that provides mapping between custom SSL oids and user-friendly names"
     },
     :basemodulepath => {
-      :default => "$codedir/modules#{File::PATH_SEPARATOR}/opt/puppetlabs/puppet/modules",
+      :default => "$codedir/modules#{File::PATH_SEPARATOR}/usr/puppetlabs/puppet/modules",
       :type => :path,
       :desc => "The search path for **global** modules. Should be specified as a
         list of directories separated by the system path separator character. (The
diff --git a/lib/puppet/face/config.rb b/lib/puppet/face/config.rb
index 19f8791b9..8707d5ec7 100644
--- a/lib/puppet/face/config.rb
+++ b/lib/puppet/face/config.rb
@@ -102,7 +102,7 @@ Puppet::Face.define(:config, '0.0.1') do
 
       Set the vardir for only the agent:
 
-      $ puppet config set vardir /opt/puppetlabs/puppet/cache --section agent
+      $ puppet config set vardir /var/puppetlabs/puppet/cache --section agent
     EOT
 
     when_invoked do |name, value, options|
diff --git a/lib/puppet/face/module/install.rb b/lib/puppet/face/module/install.rb
index 1091df730..d67c621b5 100644
--- a/lib/puppet/face/module/install.rb
+++ b/lib/puppet/face/module/install.rb
@@ -46,20 +46,21 @@ Puppet::Face.define(:module, '1.0.0') do
 
       Install a module into a specific directory:
 
-      $ puppet module install puppetlabs-vcsrepo --target-dir=/opt/puppetlabs/puppet/modules
-      Preparing to install into /opt/puppetlabs/puppet/modules ...
-      Downloading from https://forgeapi.puppet.com ...
+
+      $ puppet module install puppetlabs-vcsrepo --target-dir=/usr/puppetlabs/puppet/modules
+      Preparing to install into /usr/puppetlabs/puppet/modules ...
+      Downloading from https://forgeapi.puppetlabs.com ...
       Installing -- do not interrupt ...
-      /opt/puppetlabs/puppet/modules
+      /usr/puppetlabs/puppet/modules
       └── puppetlabs-vcsrepo (v0.0.4)
 
       Install a module into a specific directory and check for dependencies in other directories:
 
-      $ puppet module install puppetlabs-vcsrepo --target-dir=/opt/puppetlabs/puppet/modules --modulepath /etc/puppetlabs/code/modules
-      Preparing to install into /opt/puppetlabs/puppet/modules ...
-      Downloading from https://forgeapi.puppet.com ...
+      $ puppet module install puppetlabs-vcsrepo --target-dir=/usr/puppetlabs/puppet/modules --modulepath /etc/puppetlabs/code/modules
+      Preparing to install into /usr/puppetlabs/puppet/modules ...
+      Downloading from https://forgeapi.puppetlabs.com ...
       Installing -- do not interrupt ...
-      /opt/puppetlabs/puppet/modules
+      /usr/puppetlabs/puppet/modules
       └── puppetlabs-vcsrepo (v0.0.4)
 
       Install a module from a release archive:
diff --git a/lib/puppet/face/module/list.rb b/lib/puppet/face/module/list.rb
index cecaf5b20..df51df626 100644
--- a/lib/puppet/face/module/list.rb
+++ b/lib/puppet/face/module/list.rb
@@ -27,7 +27,7 @@ Puppet::Face.define(:module, '1.0.0') do
         ├── puppetlabs-mysql (v0.0.1)
         ├── puppetlabs-sqlite (v0.0.1)
         └── puppetlabs-stdlib (v2.2.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/puppetlabs/puppet/modules (no modules installed)
 
       List installed modules in a tree view:
 
@@ -38,7 +38,7 @@ Puppet::Face.define(:module, '1.0.0') do
           ├─┬ puppetlabs-mysql (v0.0.1)
           │ └── bodepd-create_resources (v0.0.1)
           └── puppetlabs-sqlite (v0.0.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/puppetlabs/puppet/modules (no modules installed)
 
       List installed modules from a specified environment:
 
@@ -49,12 +49,12 @@ Puppet::Face.define(:module, '1.0.0') do
         ├── puppetlabs-mysql (v0.0.1)
         ├── puppetlabs-sqlite (v0.0.1)
         └── puppetlabs-stdlib (v2.2.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/puppetlabs/puppet/modules (no modules installed)
 
       List installed modules from a specified modulepath:
 
-      $ puppet module list --modulepath /opt/puppetlabs/puppet/modules
-        /opt/puppetlabs/puppet/modules (no modules installed)
+      $ puppet module list --modulepath /usr/puppetlabs/puppet/modules
+        /usr/puppetlabs/puppet/modules (no modules installed)
     EOT
 
     when_invoked do |options|
diff --git a/lib/puppet/face/module/uninstall.rb b/lib/puppet/face/module/uninstall.rb
index e3c74c53c..90c6ed4db 100644
--- a/lib/puppet/face/module/uninstall.rb
+++ b/lib/puppet/face/module/uninstall.rb
@@ -16,8 +16,8 @@ Puppet::Face.define(:module, '1.0.0') do
 
       Uninstall a module from a specific directory:
 
-      $ puppet module uninstall puppetlabs-ssh --modulepath /opt/puppetlabs/puppet/modules
-      Removed /opt/puppetlabs/puppet/modules/ssh (v1.0.0)
+      $ puppet module uninstall puppetlabs-ssh --modulepath /usr/puppetlabs/puppet/modules
+      Removed /usr/puppetlabs/puppet/modules/ssh (v1.0.0)
 
       Uninstall a module from a specific environment:
 
diff --git a/lib/puppet/provider/package/puppet_gem.rb b/lib/puppet/provider/package/puppet_gem.rb
index 8d876a06a..86752b74a 100644
--- a/lib/puppet/provider/package/puppet_gem.rb
+++ b/lib/puppet/provider/package/puppet_gem.rb
@@ -12,6 +12,6 @@ Puppet::Type.type(:package).provide :puppet_gem, :parent => :gem do
     # sort out the absolute path.
     commands :gemcmd => "gem"
   else
-    commands :gemcmd => "/opt/puppetlabs/puppet/bin/gem"
+    commands :gemcmd => "/usr/bin/gem"
   end
 end
diff --git a/lib/puppet/reference/configuration.rb b/lib/puppet/reference/configuration.rb
index dcac2d0ff..5e7d25ce9 100644
--- a/lib/puppet/reference/configuration.rb
+++ b/lib/puppet/reference/configuration.rb
@@ -23,7 +23,7 @@ config = Puppet::Util::Reference.newreference(:configuration, :depth => 1, :doc
     # Now print the data about the item.
     val = object.default
     if name.to_s == 'vardir'
-      val = 'Unix/Linux: /opt/puppetlabs/puppet/cache -- Windows: C:\ProgramData\PuppetLabs\puppet\cache -- Non-root user: ~/.puppetlabs/opt/puppet/cache'
+      val = 'Unix/Linux: /var/puppetlabs/puppet/cache -- Windows: C:\ProgramData\PuppetLabs\puppet\cache -- Non-root user: ~/.puppetlabs/opt/puppet/cache'
     elsif name.to_s == 'confdir'
       val = 'Unix/Linux: /etc/puppetlabs/puppet -- Windows: C:\ProgramData\PuppetLabs\puppet\etc -- Non-root user: ~/.puppetlabs/etc/puppet'
     elsif name.to_s == 'codedir'
diff --git a/lib/puppet/util/run_mode.rb b/lib/puppet/util/run_mode.rb
index cf6c4613a..21968cea8 100644
--- a/lib/puppet/util/run_mode.rb
+++ b/lib/puppet/util/run_mode.rb
@@ -67,7 +67,7 @@ module Puppet
       end
 
       def var_dir
-        which_dir("/opt/puppetlabs/puppet/cache", "~/.puppetlabs/opt/puppet/cache")
+        which_dir("/var/puppetlabs/puppet/cache", "~/.puppetlabs/opt/puppet/cache")
       end
 
       def run_dir
diff --git a/man/man8/puppet-config.8 b/man/man8/puppet-config.8
index 35310d0c4..5df37e4da 100644
--- a/man/man8/puppet-config.8
+++ b/man/man8/puppet-config.8
@@ -111,7 +111,7 @@ $ puppet config set rundir /var/run/puppetlabs
 Set the vardir for only the agent:
 .
 .P
-$ puppet config set vardir /opt/puppetlabs/puppet/cache \-\-section agent
+$ puppet config set vardir /var/puppetlabs/puppet/cache \-\-section agent
 .
 .SH "COPYRIGHT AND LICENSE"
 Copyright 2011 by Puppet Labs Apache 2 license; see COPYING
diff --git a/spec/unit/provider/package/puppet_gem_spec.rb b/spec/unit/provider/package/puppet_gem_spec.rb
index 017e0c408..2817701dd 100644
--- a/spec/unit/provider/package/puppet_gem_spec.rb
+++ b/spec/unit/provider/package/puppet_gem_spec.rb
@@ -20,7 +20,7 @@ describe provider_class do
   if Puppet.features.microsoft_windows?
     let(:puppet_gem) { 'gem' }
   else
-    let(:puppet_gem) { '/opt/puppetlabs/puppet/bin/gem' }
+    let(:puppet_gem) { '/usr/bin/gem' }
   end
 
   before :each do
diff --git a/spec/unit/util/run_mode_spec.rb b/spec/unit/util/run_mode_spec.rb
index bd7759a4e..657d40c80 100644
--- a/spec/unit/util/run_mode_spec.rb
+++ b/spec/unit/util/run_mode_spec.rb
@@ -67,8 +67,8 @@ describe Puppet::Util::RunMode do
     end
 
     describe "#var_dir" do
-      it "has vardir /opt/puppetlabs/puppet/cache when run as root" do
-        as_root { expect(@run_mode.var_dir).to eq(File.expand_path('/opt/puppetlabs/puppet/cache')) }
+      it "has vardir /var/puppetlabs/puppet/cache when run as root" do
+        as_root { expect(@run_mode.var_dir).to eq(File.expand_path('/var/puppetlabs/puppet/cache')) }
       end
 
       it "has vardir ~/.puppetlabs/opt/puppet/cache when run as non-root" do
-- 
2.14.1


